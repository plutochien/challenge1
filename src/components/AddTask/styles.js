import styled from 'styled-components';

export const Input = styled.input`
  width: 80%;
  border: 0;
  height: 40px;
  border-radius: 5px;
  text-indent: 10px;
  outline: none;
  font-size:20px;
}
`;

export const Button = styled.button`
  margin-left: 10px;
  padding: 8px 20px 8px 20px;
  margin: 0 20px 0 20px;
  background: #007bff;
  color: #fff;
  border-radius: 3px;
  border: 2px solid #007bff;
  outline: none;
  cursor: pointer;
  font-size: 20px;
`;
