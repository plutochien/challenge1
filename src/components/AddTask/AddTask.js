import PropTypes from 'prop-types';
import React, { Fragment,useEffect, useRef, useState } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { addTask } from '../../actions/todoAction';
import {ADD, ADD_TASK, PRIORITY, PRIORITY_VALUE} from '../../util/constatns';
import { Button, Input, Label, RaidoGroup } from './styles';

const AddTask = ({ addTask }) => {
  const inputRef = useRef(null);
  const [priority, setPriority] = useState(PRIORITY_VALUE.PRIORITY_MEDIUM);
  useEffect(() => {
    if (inputRef && inputRef.current) {
      inputRef.current.focus();
    }
  }, []);

  const handleSubmitButton = event => {
    event.preventDefault();
    const taskText = inputRef.current.value.trim();
    if (!taskText) {
      return;
    }
    addTask({ taskText, priority });
    inputRef.current.value = '';
    inputRef.current.focus();
    setPriority(PRIORITY_VALUE.PRIORITY_MEDIUM);
  };

  const handleRadioButton = event => {
    setPriority(event.target.value);
  };

  return (
    <Fragment>
      <Input ref={inputRef} placeholder={ADD_TASK} />
      <RaidoGroup>
        <Input
          name="priority"
          type="radio"
          checked={priority === PRIORITY_VALUE.PRIORITY_URGENT}
          value={PRIORITY_VALUE.PRIORITY_URGENT}
          onChange={handleRadioButton}
        />
        <Label priority={PRIORITY_VALUE.PRIORITY_URGENT}>{PRIORITY.PRIORITY_URGENT}</Label>
        <Input
          name="priority"
          type="radio"
          checked={priority === PRIORITY_VALUE.PRIORITY_MEDIUM}
          value={PRIORITY_VALUE.PRIORITY_MEDIUM}
          onChange={handleRadioButton}
        />
        <Label>{PRIORITY.PRIORITY_MEDIUM}</Label>
        <Input
          name="priority"
          type="radio"
          checked={priority === PRIORITY_VALUE.PRIORITY_LOW}
          value={PRIORITY_VALUE.PRIORITY_LOW}
          onChange={handleRadioButton}
        />
        <Label priority={PRIORITY_VALUE.PRIORITY_LOW}>{PRIORITY.PRIORITY_LOW}</Label>
      </RaidoGroup>
      <Button type="submit" onClick={handleSubmitButton}>
        {ADD}
      </Button>
    </Fragment>
  );
};

AddTask.propTypes = {
  addTask: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      addTask,
    },
    dispatch,
  );

export default connect(null, mapDispatchToProps)(AddTask);
