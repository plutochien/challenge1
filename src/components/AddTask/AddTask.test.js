import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import AddTask from './AddTask';
const mockStore = configureStore([]);

describe('AddTask Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      todos: {
        list: [
          {
            id: 7,
            taskText: 'task1',
            isComplete: true,
          },
          {
            id: 6,
            taskText: 'task2',
            isComplete: false,
          },
          {
            id: 1,
            taskText: 'task3',
            isComplete: true,
          },
        ],
      },
      filter: 'SHOW_ALL',
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <AddTask />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  // it('should dispatch an action on click Add task button', () => {
  //   renderer.act(() => {
  //     component.root.findByType('button').props.onClick();
  //   });
  //   expect(store.dispatch).toHaveBeenCalledTimes(2);
  // });
});
