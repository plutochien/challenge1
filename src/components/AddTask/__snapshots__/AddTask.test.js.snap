// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask Component should render with given state from Redux store 1`] = `
Array [
  <input
    className="sc-bdfBwQ dQhNnk"
    placeholder="Add task"
  />,
  <div
    className="sc-dlfnbm ggRcYP"
  >
    <input
      checked={false}
      className="sc-bdfBwQ eywrwi"
      name="priority"
      onChange={[Function]}
      type="radio"
      value="0"
    />
    <label
      className="sc-hKgILt bbXJBi"
    >
      Urgent
    </label>
    <input
      checked={true}
      className="sc-bdfBwQ eywrwi"
      name="priority"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="sc-hKgILt hQUimg"
    >
      Medium
    </label>
    <input
      checked={false}
      className="sc-bdfBwQ eywrwi"
      name="priority"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="sc-hKgILt kHPCfL"
    >
      Low
    </label>
  </div>,
  <button
    className="sc-gsTCUz iWLRvp"
    onClick={[Function]}
    type="submit"
  >
    Add
  </button>,
]
`;
